packages/realm-server/tests/realm-server-test.ts

```diff
@@ -3677,6 +3678,52 @@ module('Realm Server', function (hooks) {
       assert.strictEqual(response.status, 404, 'HTTP 404 status');
     });
 
+    test('can create a user', async function (assert) {
+      let ownerUserId = '@mango:boxel.ai';
+      let response = await request2
+        .post('/_user')
+        .set('Accept', 'application/json')
+        .set('Content-Type', 'application/json')
+        .set(
+          'Authorization',
+          `Bearer ${createRealmServerJWT(
+            { user: ownerUserId, sessionRoom: 'session-room-test' },
+            secretSeed,
+          )}`,
+        )
+        .send({
+          data: {
+            type: 'user',
+            attributes: {
+              registrationToken: 'reg_token_123',
+            },
+          },
+        });
+
+      assert.strictEqual(response.status, 200, 'HTTP 200 status');
+      assert.strictEqual(response.text, 'ok', 'response body is correct');
+
+      let user = await getUserByMatrixUserId(dbAdapter, ownerUserId);
+      if (!user) {
+        throw new Error('user does not exist in db');
+      }
+      assert.strictEqual(
+        user.matrixUserId,
+        ownerUserId,
+        'matrix user ID is correct',
+      );
+      assert.strictEqual(
+        user.matrixRegistrationToken,
+        'reg_token_123',
+        'registration token is correct',
+      );
+    });
+
+    test('can not create a user without a jwt', async function (assert) {
```

Comment:
should we also have a test where the JWT does not match the user being created?

Reply:
I don't think this situation is possible–the handler will read the users's matrix id from the JWT and use that to upsert the user (with the registration token). This means the jwt will always match the user being created